Chemistry/GetGateCount/GetGateCount.cs

```diff
@@ -0,0 +1,270 @@
+        )
+        {
+            // To get the data for exporting to Q#, we proceed in several steps.
+            var jordanWignerEncoding =
+                // First, we deserialize the file given by filename, using the
+                // format given by format.
+                format.Map(
+                    (IntegralDataFormat.Liquid, () => LiQuiD.Deserialize(filename).Select(o => o.OrbitalIntegralHamiltonian)),
+                    (IntegralDataFormat.Broombridge, () => Broombridge.Deserializers.DeserializeBroombridge(filename).ProblemDescriptions.Select(o => o.OrbitalIntegralHamiltonian))
+                )
+                // In general, Broombridge allows for loading multiple Hamiltonians
+                // from a single file. For the purpose of simplicitly, however,
+                // we'll only load a single Hamiltonian from each file in this
+                // sample. We use .Single here instead of .First to make sure
+                // that we raise an error instead of silently discarding data.
+                .Single()
+                // Next, we convert to a fermionic Hamiltonian using the UpDown
+                // convention.
+                .ToFermionHamiltonian(IndexConvention.UpDown)
+                // Finally, we use the optimized Jordan–Wigner representation
+                // to convert to a qubit Hamiltonian that we can export to
+                // a format understood by Q#.
+                .ToPauliHamiltonian(Paulis.QubitEncoding.JordanWigner);
+
+            // We save the exported Q# data into a variable that we can pass
+            // to the trace simulator.
+            var qSharpData = jordanWignerEncoding
+                .ToQSharpFormat()
+                .Pad();
+
+            var gateCountResults = new List<GateCountResults>();
+
+            foreach (var config in configurations)
+            {
+                var results = await RunGateCount(qSharpData, config, outputFolder);
+                results.IntegralDataPath = filename;
+                results.SpinOrbitals = jordanWignerEncoding.SystemIndices.Count();
+                gateCountResults.Add(results);
+            }
+
+            return gateCountResults;
+        }
+
+        // This method computes the gate count of simulation a single configuration passed to it.
+        internal static async Task<GateCountResults> RunGateCount(
+            JordanWignerEncodingData qSharpData, HamiltonianSimulationConfig config,
+            string outputFolder = null
+        )
+        {
+            // Creates and configures Trace simulator for accumulating gate counts.
+            var sim = CreateAndConfigureTraceSim();
+
+            // Create stop-watch to measure the execution time
+            var stopWatch = new Stopwatch();
+
+            var gateCountResults = new GateCountResults();
+            #region Trace Simulator on Trotter step
+            if (config.hamiltonianSimulationAlgorithm == HamiltonianSimulationAlgorithm.ProductFormula)
```

Comment:
Each of these are mostly copy/paste... I would take the Configuration concept and go a step further and move the which Q# operation to run as part of the configuration and a `Name` property, so you don't have to have the code duplication here.

Reply:
Agreed. I actually tried to do that, but ran into a couple slight issues in that the `GetMetric` method of the trace simulator takes a type argument for the operation in question, and in that the optimized qubitization operation takes an additional parameter for the target error. I'd prefer if possible to defer this improvement to the next PR on this sample if that's OK.